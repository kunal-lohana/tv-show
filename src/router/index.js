import Vue from "vue";
import VueRouter from "vue-router";
import TvShowList from "@/components/TvShowList/TvShowList.vue";
import ShowDetail from "@/components/ShowDetail/ShowDetail.vue";
import PageNotFound from "@/components/PageNotFound";

Vue.use(VueRouter);

export const routes = [
  {
    path: "/",
    component: TvShowList,
    meta: {
      title: "Tv Show"
    }
  },
  {
    path: "/:showId",
    component: ShowDetail,
    name: "showDetail",
    meta: {
      title: "Show Detail"
    }
  },
  {
    path: "*",
    name: "PageNotFound",
    component: PageNotFound,
    meta: {
      title: "Page-not-found"
    }
  }
];

export const router = new VueRouter({
  mode: "history",
  base: process.env.BASE_URL,
  routes
});

router.afterEach(to => {
  document.title = to.meta.title || "Home";
});
