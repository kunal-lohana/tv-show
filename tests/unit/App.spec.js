import { shallowMount, config, createLocalVue } from "@vue/test-utils";
import { searchData } from "./testingData";
import Vuetify from "vuetify";
import Vue from "vue";
import VueRouter from "vue-router";
import AppComponent from "../../src/App.vue";
import { routes } from "@/router/index";

config.showDeprecationWarnings = false;
Vue.config.slient = false;

describe("App Component ", () => {
  let localVue = createLocalVue();
  localVue.use(VueRouter);
  Vue.use(Vuetify);
  let vuetify;
  const router = new VueRouter({
    routes
  });
  beforeEach(() => {
    vuetify = new Vuetify();
  });

  it("should match snapshot", () => {
    const wrapper = shallowMount(AppComponent, {
      router,
      localVue,
      vuetify
    });
    expect(wrapper.html()).toMatchSnapshot();
  });

  it("Emit searchInput event will call updateSearchInput method", async () => {
    const wrapper = shallowMount(AppComponent, {
      router,
      localVue,
      vuetify
    });
    wrapper.vm.$emit("searchInput", searchData);
    await wrapper.vm.$nextTick();
    expect(wrapper.emitted("searchInput")).toBeTruthy();
  });

  it("Execute updateSearchInput method will set searchInput data prop.", async () => {
    const wrapper = shallowMount(AppComponent, {
      router,
      localVue,
      vuetify
    });
    await wrapper.vm.updateSearchInput(searchData);
    const filterData = searchData.map(data => data.show);
    expect(wrapper.vm.searchInput).toEqual(filterData);
  });
});
