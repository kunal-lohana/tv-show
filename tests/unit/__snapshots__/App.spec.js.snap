// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App Component  should match snapshot 1`] = `
<v-app-stub id="app">
  <v-main-stub tag="main" app="" dark="" class="secondary">
    <header-stub></header-stub>
    <router-view-stub name="default" searchprop=""></router-view-stub>
  </v-main-stub>
</v-app-stub>
`;
